Client/Application/AppDelegate.swift

```diff
@@ -53,7 +54,12 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     }
 
     func getProfile(application: UIApplication) -> Profile {
-        return BrowserProfile(localName: "profile", app: application)
+        if let profile = self.profile {
+            return profile
+        }
+        let p = BrowserProfile(localName: "profile", app: application)
+        self.profile = p
+        return p
```

Comment:
I don't think this works as expected. Because `self.profile` is weak, a new instance will always be created if the caller of `getProfile` does not hang on to it. For example right after the call below to `applicationDidBecomeActive`, `self.profile` will be set to `nil`.